src/discussions/comments/comment/Comment.jsx

```diff
@@ -1,50 +1,121 @@
-        <div className="d-flex flex-column flex-fill">
-          <h4 className="title">
-            {comment.title}
-          </h4>
-          <div className="status small">
-            {intl.formatMessage(messages.postTime, {
-              postType: comment.type,
+    <div className="d-flex flex-row justify-content-between">
+      <div className="align-items-center d-flex flex-row">
+        <Avatar className="m-2" alt={comment.author} src={authorAvatars?.imageUrlSmall} />
+        <div className="status small">
+          <a href="#nowhere">
+            <h1 className="font-weight-normal text-info-300 mr-1 small">
+              {comment.author} {comment.authorLabel ? `(${comment.authorLabel})` : ''}
+            </h1>
+          </a>
+          <h2 className="font-weight-normal small text-gray-500" title={comment.createdAt}>
+            {intl.formatMessage(messages.commentTime, {
               relativeTime: timeago.format(comment.createdAt, intl.locale),
             })}
-            <span className="font-weight-bold text-info-300 ml-1">{comment.author}</span>
-          </div>
-        </div>
-        <CommentIcons abuseFlagged={comment.abuseFlagged} following={comment.following} />
-      </div>
-      <div className="mt-2">
-        <div className="d-flex" dangerouslySetInnerHTML={{ __html: comment.renderedBody }} />
-        <div className="d-flex small text-gray-300">
-          {intl.formatMessage(messages.postVisibility, { group: comment.groupName })}
+          </h2>
         </div>
       </div>
+      <ActionsDropdown commentOrPost={comment} onCommentAction={onCommentAction} />
     </div>
   );
 }
 
-export const commentShape = PropTypes.shape({
-  createdAt: PropTypes.string,
-  abuseFlagged: PropTypes.bool,
-  renderedBody: PropTypes.string,
-  author: PropTypes.string,
-});
+CommentHeader.propTypes = {
+  comment: commentShape.isRequired,
+  onCommentAction: PropTypes.func.isRequired,
+  intl: intlShape.isRequired,
+};
+
+function Comment({
+  comment,
+  intl,
+}) {
+  const dispatch = useDispatch();
+  const hasChildren = !comment.parentId;
+  const inlineReplies = useSelector(selectCommentResponses(comment.id));
+  useEffect(() => {
+    // If the comment has a parent comment, it won't have any children, so don't fetch them.
+    if (hasChildren) {
+      dispatch(fetchCommentResponses(comment.id));
+    }
+  }, [comment.id]);
+
+  const handleCommentAction = (action) => {
+    switch (action) {
+      case EditActions.DELETE:
+        // TODO: add some kind of confirmation flow
+        dispatch(removeComment(comment.id));
+        break;
+      case EditActions.REPORT:
+        dispatch(updateComment(comment.id, { flagged: !comment.abuseFlagged }));
+        break;
+      case EditActions.ENDORSE:
+        dispatch(updateComment(comment.id, { endorsed: !comment.endorsed }));
+        break;
+      default:
+        console.debug(`Unexpected action ${action} on comment ${comment.id}`);
```

Comment:
Nope, the debug is there on purpose. It may be removed in the final version of the code, for but now, while not all actions are implemented it makes sense to have this here. I will replace it with a proper error log.